modules/cql/src/blaze/elm/ratio.clj

```diff
@@ -0,0 +1,52 @@
+(ns blaze.elm.ratio
+  "Implementation of the ratio type."
+  (:require
+    [blaze.elm.protocols :as p]
+    [blaze.elm.quantity :as quantity])
+  (:import (javax.measure UnconvertibleException)))
+
+
+(set! *warn-on-reflection* true)
+
+
+(defrecord Ratio [numerator denominator]
+  p/Equal
+  (equal [_ other]
+    (and (p/equal numerator (:numerator other))
+         (p/equal denominator (:denominator other))))
+
+  p/Equivalent
+  (equivalent [_ other]
+    (if (= (quantity/quantity-divide numerator denominator)
+           (quantity/quantity-divide (:numerator other) (:denominator other)))
+      (if (and (and (= (.getUnit numerator) (.getUnit denominator))
+                    (= (.getUnit (:numerator other)) (.getUnit (:denominator other))))
+               (not= (.getUnit numerator) (.getUnit (:numerator other))))
+        (try
+          (if (and (p/equal numerator (:numerator other))
+             (p/equal denominator (:denominator other)))
+          true
+          false)
+          (catch UnconvertibleException _ false))
+        true)
+      false)))
+
+
+(defn ratio
+  "Creates a ratio between two quantities."
+  [numerator denominator]
+  (->Ratio numerator denominator))
+
+
+;; 22.28. ToQuantity
+(extend-protocol p/ToQuantity
+  Ratio
+  (to-quantity [s]
+     (quantity/quantity-divide (:denominator s) (:numerator s))))
```

Comment:
Please don't use `quantity/quantity-divide` that is a private protocol for quantity. Instead please use `p/divide`.